packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -149,7 +156,21 @@ export class SequenceInterval implements ISerializableInterval {
     public compare(b: SequenceInterval) {
         const startResult = this.compareStart(b);
         if (startResult === 0) {
-            return this.compareEnd(b);
+            const endResult = this.compareEnd(b);
+            if (endResult === 0) {
+                const thisId = this.getIntervalId();
+                if (thisId) {
+                    const bId = b.getIntervalId();
+                    if (bId) {
+                        return thisId > bId ? 1 : thisId < bId ? -1 : 0;
+                    }
```

Comment:
do we explcitly support adding a interval with the same id as and existing interval for update?

Reply:
Well, we don't have an API that lets you specify the ID when you add an interval (though you could mess with an interval's ID if you wanted to once it's been created). The local add(start, end) will create a new ID, and the non-local add(serialized interval) will propagate the ID from the serialized interval. (I changed the flow so the local add(start, end) no longer gets routed through add(serialized interval).) For the case where you want to update an existing interval, I'm picturing a new API like change(ID, start, end). For now, I'm having it do a remove-update-add under the hood, but I know we want to send a single op over the wire.